core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -241,49 +234,52 @@ oppia.factory('TopicEditorStateService', [
 
       /**
        * Sets the updated subtopic page object in the correct position in the
-       * _subtopicPages list.
+       * _cachedSubtopicPages list.
        */
       setSubtopicPage: function(subtopicPage) {
         if (_getSubtopicPageIndex(subtopicPage.getId()) !== null) {
-          _subtopicPages[_getSubtopicPageIndex(subtopicPage.getId())] =
+          _cachedSubtopicPages[_getSubtopicPageIndex(subtopicPage.getId())] =
             angular.copy(subtopicPage);
           _subtopicPage.copyFromSubtopicPage(subtopicPage);
         } else {
           _setSubtopicPage(subtopicPage);
-          _newSubtopicPages.push(angular.copy(subtopicPage));
+          _newSubtopicPageIds.push(subtopicPage.getId());
         }
       },
 
       deleteSubtopicPage: function(topicId, subtopicId) {
         var subtopicPageId = _getSubtopicPageId(topicId, subtopicId);
         var index = _getSubtopicPageIndex(subtopicPageId);
-        var newIndex = _getNewSubtopicPageIndex(subtopicPageId);
+        var newIndex = _newSubtopicPageIds.indexOf(subtopicPageId);
+        // If index is null, that means the corresponding subtopic page was
+        // never loaded from the backend and not that the subtopic page doesn't
+        // exist at all. So, not required to throw an error here.
```

Comment:
Should include some explanation of why we don't need to modify newSubtopicPageIds either.

Though you might want to do that by instead adding an _assertion_ that the ID isn't in newSubtopicPageIds (else error). Safer. Should still have explanatory comments in either case though.

Reply:
Done, added check and comment.